### PR TITLE
Add instance url scopeModel

### PR DESCRIPTION
**Pull request by CrazyHZM**
## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Review by AlbumenJ — APPROVED** [review]

File: dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/filter/DefaultFilterChainBuilder.java [hunk 1]
```diff
@@ -36,7 +40,17 @@
     public <T> Invoker<T> buildInvokerChain(final Invoker<T> originalInvoker, String key, String group) {
         Invoker<T> last = originalInvoker;
         URL url = originalInvoker.getUrl();
-        List<Filter> filters = ScopeModelUtil.getExtensionLoader(Filter.class, url.getScopeModel()).getActivateExtension(url, key, group);
+        List<ModuleModel> moduleModels = getModuleModelsFromUrl(url);
+        List<Filter> filters = new ArrayList<>();
+        if (moduleModels != null && moduleModels.size() != 0) {
```
Comment (AlbumenJ) [inline dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/filter/DefaultFilterChainBuilder.java:45]: If moduleModels' size is 1, directly use result to reduce memory allocation.

File: dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/filter/DefaultFilterChainBuilder.java [hunk 2]
```diff
@@ -36,7 +40,17 @@
...
+            for (ModuleModel moduleModel : moduleModels) {
+                List<Filter> tempFilters = ScopeModelUtil.getExtensionLoader(Filter.class, moduleModel).getActivateExtension(url, key, group);
+                filters.addAll(tempFilters);
```
Comment (AlbumenJ) [inline dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/filter/DefaultFilterChainBuilder.java:48]: Filters need to be sorted and de-duplicated.

File: dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/filter/DefaultFilterChainBuilder.java [hunk 3]
```diff
@@ -69,4 +92,16 @@
         return last;
     }
 
+    private List<ModuleModel> getModuleModelsFromUrl(URL url) {
+        List<ModuleModel> moduleModels = null;
+        ScopeModel scopeModel = url.getScopeModel();
+        if (scopeModel instanceof ApplicationModel) {
+            moduleModels = ((ApplicationModel) scopeModel).getModuleModels();
```
Comment (AlbumenJ) [inline dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/filter/DefaultFilterChainBuilder.java:99]: Use `getPubModuleModels()` instead.

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

**Comment by chickenlj** [comment]
https://github.com/apache/dubbo/issues/9403

**Comment by chickenlj** [comment]
I think `getModuleModelsFromUrl` is a little bit weird. It can be better if the module model can be passed through the API method.

**Pull request merged**